app/classifier/tasks/text/index.jsx

```diff
@@ -0,0 +1,189 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'lodash';
+
+import GenericTask from '../generic';
+import TextTaskEditor from './editor';
+import TextTaskSummary from './summary';
+
+const LINEHEIGHT = 22.5;
+const NOOP = Function.prototype;
+
+export default class TextTask extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      rows: 1,
+      value: ''
+    };
+
+    this.textInput = React.createRef();
+
+    this.debouncedUpdateAnnotation = _.debounce(this.updateAnnotation, 500).bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+    this.setTagSelection = this.setTagSelection.bind(this);
+    this.setValue = this.setValue.bind(this);
+    this.updateAnnotation = this.updateAnnotation.bind(this);
+  }
+
+  componentDidMount() {
+    this.setValue();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.task && this.props.task && (prevProps.task !== this.props.task)) {
+      const value = prevState.value;
+      const newAnnotation = Object.assign(prevProps.annotation, { value });
+      prevProps.onChange(newAnnotation);
+
+      this.setValue();
+    }
+
+    this.handleResize();
+  }
+
+  componentWillUnmount() {
+    this.updateAnnotation();
+  }
+
+  setTagSelection(e) {
+    const textTag = e.target.value;
+    const startTag = `[${textTag}]`;
+    const endTag = `[/${textTag}]`;
+    const textArea = this.textInput.current;
+    const textAreaValue = textArea.value;
+    const selectionStart = textArea.selectionStart;
+    const selectionEnd = textArea.selectionEnd;
+    const textBefore = textAreaValue.substring(0, selectionStart);
+    let textAfter;
+    let value;
+    if (selectionStart === selectionEnd) {
+      textAfter = textAreaValue.substring(selectionStart, textAreaValue.length);
+      value = textBefore + startTag + endTag + textAfter;
+    } else {
+      const textInBetween = textAreaValue.substring(selectionStart, selectionEnd);
+      textAfter = textAreaValue.substring(selectionEnd, textAreaValue.length);
+      value = textBefore + startTag + textInBetween + endTag + textAfter;
+    }
+    this.setState({ value });
+    this.debouncedUpdateAnnotation();
+  }
+
+  setValue() {
```

Comment:
should prob be renamed, maybe something like `setInitialValues` or open to suggestion

Reply:
If you're setting state from props, React 16 now has [`TextTask.getDerivedStateFromProps()`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops), which might be what you're looking for.

Reply:
I tried using `getDerivedStateFromProps`, and could set the initial state value as the provided props annotation value, but had issues when clicking a tag. I'll take another look today, I thought I tried it, but maybe if `setTagSelection` calls `updateAnnotation` `getDerivedStateFromProps` can work as intended.

Reply:
`getDerivedStateFromProps` runs after every update. Looking the alternatives mentioned in the docs, the case here might be this one:
>If you want to “reset” some state when a prop changes, consider either making a component fully controlled or fully uncontrolled with a key instead.

Reply:
If you look at [the uncontrolled example](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key), it turns out you could use a key to force React to remount the task component when a new task loads. Maybe use the task key from the workflow?

I think that might simplify the case where the task changes but the component is already mounted.

Reply:
Initial attempt without luck, replaced constructor state with state declaration like in the uncontrolled example, added key in textarea of this.props.annotation.task, but with back to back text tasks the annotation would carryover to the second task. Will look again with fresh eyes on Wednesday.

Reply:
>added key in textarea of this.props.annotation.task
I think the key needs to be added to the task component itself. I've tried that in #4745.